actionpack/test/controller/action_pack_assertions_test.rb

```diff
@@ -170,6 +170,16 @@ def test_string_constraint
     end
   end
 
+  def test_with_routing_works_with_api_only_controllers
+    @controller = Class.new(ActionController::API).new
+
+    with_routing do |set|
+      set.draw do
+        get "wow", to: "action_pack_assertions#nothing"
+      end
```

Comment:
I'd add a `process` and `assert` like what we do here https://github.com/yukideluxe/rails/blob/8e9c8a19afc3f41b5bb9596ad604c7f386a903d0/actionpack/test/controller/action_pack_assertions_test.rb#L220